data-raw/admiral_verify_templates.R

```diff
@@ -0,0 +1,354 @@
+
+  # nolint start
+  library(pkg, character.only = TRUE)
+  library(teal.data)
+  library(purrr)
+  library(cli)
+  # nolint end
+  cli_alert("Generating ADaMs for { pkg} package.")
+
+  # TODO:  do not show Warnings
+  # creae temporary directories
+  cli_inform("Creating temporary directories")
+  x <- tempdir()
+  dir.create(file.path(x, "old"), showWarnings = TRUE)
+  dir.create(file.path(x, "new"), showWarnings = TRUE)
+  dir.create(file.path(x, "diff"), showWarnings = TRUE)
+
+  # TODO:
+  path <- list(
+    template_dir = "inst/templates",
+    cache_dir = tools::R_user_dir("admiral_templates_data", which = "cache"),
+    adam_new_dir = file.path(x, "new"),
+    adam_old_dir = file.path(x, "old"),
+    diff = file.path(x, "diff")
+  )
+
+  # TODO:
+  # if dir exists then empty it ; if not exist then create it.
+  # lapply(path, function(x)  if( x %in% c("template_dir", "cache_dir")!exists(x)) dir.create(x))
+
+  # gather all templates for this pkg (12 found) ----
+  templates <- list.files(path$template_dir, pattern = "ad_")
+
+  # from templates  generate vector of adam_names
+  adam_names <- vapply(templates, function(x) gsub("ad_|\\.R", "", x),
+    USE.NAMES = FALSE, character(length = 1)
+  )
+
+  # check
+  if (length(templates) != length(adam_names)) stop("Number of templates and adam_names differ")
+
+  # templates is a named chr[]
+  names(templates) <- adam_names
+
+  # User specified templates in `ds`. Limit to only these templates.
+
+  adam_names <- adam_names[adam_names %in% ds]
+
+  # per Ben, ignore "adlbhy"
+  adam_names <- adam_names[adam_names != "adlbhy"]
+
+  # download, save prior ADaMs from pharmaverseadam as .rda files
+  cli_inform("Downloading from github pharmaverseadam")
+  download_adam_old(adam_names, path = path$adam_old_dir)
+
+  # keys for diffdf
+  keys <- teal.data::default_cdisc_join_keys
+
+  # keys is named list of keys, each element (for each adam_name) and is chr[] of keys
+  keys <- sapply(toupper(adam_names), function(e) unname(keys[[e]][[e]]), # nolint
+    USE.NAMES = TRUE, simplify = FALSE
+  )
+  names(keys) <- tolower(names(keys))
+
+  # finally, construct a named list object: each element (adam_names) holds a template and keys
+  obj <- lapply(adam_names, function(e) {
+    list(
+      "template" = templates[[e]],
+      "keys" = keys[[e]]
+    )
+  })
+  names(obj) <- adam_names
+
+  sprintf("---- Run templates\n")
+  compare_list <- purrr::map(adam_names, .progress = TRUE, function(adam) {
+    cli_inform("Template running for adam")
+    run_template(adam, dir = path$template_dir)
+    # retrieve *.rda file in cache; copy to correct directory
+    dataset_new = load_rda(paste0(path$cache_dir, "/", adam, ".rda"))
+    file.copy(file.path(path$cache_dir, adam, ".rda"),
+              file.path(path$adam_new_dir, adam, ".rda"))
+    dataset_old <- get_dataset_old(adam, path$adam_old_dir)
+    compare(
+      base = dataset_old,
+      compare = dataset_new,
+      keys = obj[[adam]]$keys,
```

Comment:
I think this should work, and will allow you to remove some code above.
```suggestion
      keys = teal.data::default_cdisc_join_keys[[adam]],
```